packages/python/plotly/plotly/express/_core.py

```diff
@@ -1007,6 +1007,99 @@ def build_dataframe(args, attrables, array_attrables):
     return args
 
 
+def _discrete_agg(x):
+    if len(x) == 1:
+        return x.iloc[0]
+    else:
+        return ""
+
+
+def process_dataframe_hierarchy(args):
+    """
+    Build dataframe for sunburst or treemap when the path argument is provided.
+    """
+    df = args["data_frame"]
+    path = args["path"][::-1]
+
+    # ------------ Define aggregation functions --------------------------------
+    lambda_discrete = _discrete_agg
+    agg_f = {}
+    aggfunc_color = None
+    if args["values"]:
+        try:
+            df[args["values"]] = pd.to_numeric(df[args["values"]])
+        except ValueError:
+            raise ValueError(
+                "Column `%s` of `df` could not be converted to a numerical data type."
+                % args["values"]
+            )
+
+        if args["color"]:
+            if args["color"] == args["values"]:
+                aggfunc_color = "sum"
```

Comment:
this confuses me a little bit... can you explain?

Reply:
I think doing the weighted-average still makes sense to me here... if I have two children of value=color=[2 and 3], then my value is 5 but my color should be (3*3+2*2)/(3+2) = 2.6

Reply:
hmmm now that I write it out, this makes no sense. `sum` is the better approach here indeed.